### ISSUE TITLE
[RadChart] Issue with setting the ChartGridLineAnnotation value, while using ngIf directive

### ISSUE THREAD

**Issue by tsonevn** [telerik/nativescript-ui-feedback#296]
[Archive.zip](https://github.com/telerik/nativescript-ui-feedback/files/1314677/Archive.zip)
The problem could be reproduced with the attached project while selecting the tabs as follow  Day->Month->Year-> and then return back to Month. Then the content for the chart will not be shown as well as the annotation. This problem is reproduced only for Android.

As a workaround, we could use `visibility` property instead of `ngIf`.
```
<GridLayout height="300" rows="auto, *">
<RadCartesianChart  [visibility]="section!='year' ? 'visible' : 'collapsed'" height="250" row="0" tkExampleTitle tkToggleNavButton>
    <LinearAxis tkCartesianVerticalAxis></LinearAxis>

    <DateTimeContinuousAxis #myDateTimeContinuousAxis tkCartesianHorizontalAxis id="hAxis" [minimum]="minimum" [maximum]="maximum" 
            [majorStep]="majorStep" [dateFormat]="dateFormat" labelFitMode="Rotate" labelRotationAngle="0.9"></DateTimeContinuousAxis>
    <LineSeries tkCartesianSeries [items]="categoricalSource" categoryProperty="date" valueProperty="amount"></LineSeries>
    <ChartGridLineAnnotation tkCartesianAnnotations axisId="hAxis" hidden="false" 
                            zPosition="BelowSeries" strokeWidth="4" [value]="getAnnotation"
                            strokeColor="#EB916580">
    </ChartGridLineAnnotation>
    
    
</RadCartesianChart>


<RadCartesianChart [visibility]="section=='year' ? 'visible' : 'collapsed'" height="250" row="1" tkExampleTitle tkToggleNavButton>
    <LinearAxis tkCartesianVerticalAxis></LinearAxis>
    <DateTimeCategoricalAxis tkCartesianHorizontalAxis dateFormat="yyyy" ></DateTimeCategoricalAxis>
    <BarSeries tkCartesianSeries [items]="categoricalSource" categoryProperty="date" valueProperty="amount"></BarSeries>

</RadCartesianChart>
</GridLayout>
```

**Comment by gbonline** [comment]
Hi, The workaround is valid only for android, on ios Simulator it crash the app with segmentation fault: 11

**Comment by NickIliev** [comment]
Reported via t.1136969

**Comment by gbonline** [comment]
Hi Nickiliev, I wrote a little demo with the issues, attached to this message.
On app folder, on the file "chart-series-line.component.html"
there are the tags RadCartesianChart : one for the linear graph and other for the bar chart.
the first case have the *ngIf property and a second commented have the visibility property
try it
Giorgio
[graphdemo 2.zip](https://github.com/telerik/nativescript-ui-feedback/files/1421313/graphdemo.2.zip)

**Comment by zbranzov** [comment]
The issues are fixed in nativescript-ui-chart version 3.6. Enjoy!
